src/common/heartbeat/src/listener.cpp

```diff
@@ -1,26 +1,138 @@
+#include <iostream>
+#include <sstream>
+
 #include <rclcpp/rclcpp.hpp>
-#include <std_msgs/msg/bool.hpp>
+#include <builtin_interfaces/msg/time.hpp>
+#include <std_msgs/msg/empty.hpp>
 
-#include <iostream>
+#include "common_interfaces/msg/e_stop.hpp"
+#include "common_interfaces/srv/handshake.hpp"
+#include "common_interfaces/srv/heartbeat_control.hpp"
+#include "heartbeat/date.h"
+
+// TODO: ros params instead of this
+#define PERIOD 3000 // how long should the time between pings be? (negotiated during handshake)
```

Comment:
Yeah that's a "definitely". The watchdog_ms can remain a #define, I think it's unlikely that we would want to change that one. Then again, if you're making it so you can change the other two, you might as well change WATCHDOG_MS too. 
You can either implement this parameterize as a once-at-startup thing or a dynamically changeable thing.... thing is, since these fields are being used as the period for timer objects, you would need to drop and re-create those timers in the ROS on-parameter-change (whatever it's called) function, which is not so great. I would say restarting the robot to change this parameter is still preferable enough to rebuilding everything to make it worth it.